sktime/datasets/_fpp3_loaders.py

```diff
@@ -0,0 +1,274 @@
+    fpp3 = ["aus_accommodation", "aus_airpassengers", "aus_arrivals", "bank_calls", \
+               "boston_marathon", "canadian_gas", "guinea_rice", "insurance", "prices", \
+               "souvenirs", "us_change", "us_employment", "us_gasoline"]
+    tsibble = ["pedestrian", "tourism"]
+
+    tsibbledata = ["ansett", "aus_livestock", "aus_production", "aus_retail", "gafa_stock", \
+                   "global_economy", "hh_budget", "nyc_bikes", "olympic_running", "PBS", "pelt", \
+                   "vic_elec"]
+
+    url_fpp3 = "https://cran.r-project.org/src/contrib/fpp3_0.5.tar.gz"
+    url_tsibble = "https://cran.r-project.org/src/contrib/tsibble_1.1.4.tar.gz"
+    url_tsibbledata = "https://cran.r-project.org/src/contrib/tsibbledata_0.4.1.tar.gz"
+
+    if dataset_name in fpp3:
+        return (True, url_fpp3)
+    if dataset_name in tsibble:
+        return (True, url_tsibble)
+    if dataset_name in tsibbledata:
+        return (True, url_tsibbledata)
+
+    return (False, None)
+
+def _decompress_file_to_temp(url, temp_folder="/tmp"):
+    import requests    
+    temp_dir = tempfile.mkdtemp(dir=temp_folder)
+    response = requests.get(url)
+    temp_file = os.path.join(temp_dir, "foo.tar.gz")
+    with open(temp_file, 'wb') as f:
+        f.write(response.content)
+    tar = tarfile.open(temp_file)
+    tar.extractall(path=temp_dir)
+    tar.close()
+    return temp_dir
+
+def _find_dataset(temp_folder, dataset_name):
+    dataset = dataset_name + ".rda"
+    for root, dirs, files in os.walk(temp_folder):
+        if dataset in files:
+            return (True, os.path.join(root, dataset))
+    return (False, None)
+
+def _yearweek_constructor(obj, attrs):
+    return pd.to_datetime(obj, origin='1970-01-01', unit='D').to_period('W').astype(str)
+
+def _yearmonth_constructor(obj, attrs):
+    return pd.to_datetime(obj, origin='1970-01-01', unit='D').to_period('M').astype(str)
+
+def _yearquarter_constructor(obj, attrs):
+    return pd.to_datetime(obj, origin='1970-01-01', unit='D').to_period('Q').astype(str)
+
+def _date_constructor(obj, attrs):
+    return pd.to_datetime(obj, origin='1970-01-01', unit='D')
+
+def _import_rda(path):
+    import rdata
+
+    constructor_dict = {
+        **rdata.conversion.DEFAULT_CLASS_MAP,
+        "Date": _date_constructor,
+        "yearweek": _yearweek_constructor,
+        "yearmonth": _yearmonth_constructor,
+        "yearquarter": _yearquarter_constructor,
+    }
+    show_warnings = False
+    if show_warnings:
+        obj = rdata.read_rda(path, constructor_dict=constructor_dict)
+    else:
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            obj = rdata.read_rda(path, constructor_dict=constructor_dict)
+
+    if len(obj) == 1:
+        return (True, next(iter(obj.values())))
+
+    return (False, obj)
+
+def _dataset_to_mtype(dataset_name, obj):
+    if dataset_name in ["aus_airpassengers", "guinea_rice", "pelt", "prices", "olympic_running", \
```

Comment:
looks like a lot of work to get this function working!

Design question: can this be refactored into a "data format card" or similar, where you have, per dataset, a small number of configs, and then you look up the configs and carry out the steps? I do not see a clear pattern, so perhaps it's sth for another PR, or to make a note of in an issue.

Reply:
I did something like this in the hosted data version (previous PR that was rejected to keep the size of sktime to a minimum.) I created a dictionary that contained various parameters and even lambda functions for pre- and post-processing. That was a little trickier here because rdata.read_rda() has its own default parsing, although you can extend the parser to handle unrecognized R classes (some of which were present and which I did add.) I think there have been enough hurdles that I have had to jump to get this far. Alternatively, while the code is not particularly compact it is easy to debug :-)

Reply:
sorry for the hurdles, we're always trading-off user hurdles with developer hurdles and like to err on the side of users... your patience is much appreciated, and given that you are also a user, I hope you second this policy as well.